app/scripts/lib/crypto/recovery-keys.js

```diff
@@ -0,0 +1,151 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * This file contains utilities to help create, bundle and unbundle encrypted
+ * recovery key data.
+ *
+ * For more encryption details, check out
+ + https://github.com/mozilla/fxa-auth-server/blob/master/docs/recovery_keys.md
+ */
+const Base32 = require('./base32');
+
+function importFxaCryptoDeriver() {
+  return import(/* webpackChunkName: "fxaCryptoDeriver" */ 'fxaCryptoDeriver');
+}
+
+function required(object, name) {
+  if (! object) {
+    throw new Error(`Missing ${name}`);
+  }
+}
+
+const RECOVERY_KEY_VERSION = '\x01';
+const ENCRYPTION_ALGORITHM = 'A256GCM';
+const HKDF_SHA_256 = 'HKDF-SHA-256';
+
+module.exports = {
+  /**
+   * Generate a random base32 recovery key. The recovery key
+   * string is prepended with version information.
+   *
+   * @param {Integer} length Length of string to generate
+   * @returns {Promise<string>} recovery key
+   */
+  generateRecoveryKey: function (length) {
+    return Base32.generate(length - 1)
+      .then((key) => RECOVERY_KEY_VERSION + key);
+  },
+
+  /**
+   * Retrieves the recovery JWK from the uid and recovery key. The
+   * recoveryJwk can be used to encrypt and decrypt data.
+   *
+   * @param {String} uid Uid of user
+   * @param {String} recoveryKey Recovery key
+   * @returns {Promise} A promise that will be fulfilled with JWK
+   */
+  getRecoveryJwk: function (uid, recoveryKey) {
+    return importFxaCryptoDeriver().then(({jose}) => {
+      required(uid, 'uid');
+      required(recoveryKey, 'recoveryKey');
+
+      return Base32.decode(recoveryKey)
```

Comment:
has this been verified with output from `generateRecoveryKey()`?  I suspect prepending the binary RECOVERY_KEY_VERSION may cause an error ...

Reply:
Note: I think `recoveryKey` should include the version always, meaning at most check the version matches expectations but don't strip it when performing the KDFs.